CMakeLists.txt

```diff
@@ -0,0 +1,59 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(SugarPP
+        VERSION 0.0.1
+        HOMEPAGE_URL "https://github.com/HO-COOH/SugarPP"
+        LANGUAGES CXX)
+
+# ---- Warning guard ----
+
+# Protect dependents from this project's warnings if the guard isn't disabled
+set(sugarpp_warning_guard "SYSTEM")
+if(SugarPP_INCLUDE_WITHOUT_SYSTEM)
+  set(sugarpp_warning_guard "")
+endif()
+
+# ---- Declare library ----
+
+add_library(SugarPP INTERFACE)
+add_library(SugarPP::SugarPP ALIAS SugarPP)
+
+target_include_directories(SugarPP
+        ${sugarpp_warning_guard}
+        INTERFACE
+        "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>")
```

Comment:
Yes, populating `INTERFACE_INCLUDE_DIRECTORIES` is handled by the `install(TARGETS ... INCLUDES DESTINATION ...)` special form, so that generator expression is unnecessary here.

Reply:
Cool, didn't know that.